tools/pvacvector/run.py

```diff
@@ -105,54 +105,66 @@ def run_pipelines(input_file, base_output_dir, args):
     return parsed_output_files
 
 def find_min_scores(parsed_output_files, args):
-    iedb_results = {}
-    epitopes = []
-    indexes = []
+    min_scores = {}
+    indexes_with_good_binders = []
+    #find indexes that contain a good binder so that they can be excluded from further processing
+    #we don't want any peptide-spacer-peptide combination (aka index) that contains a good binder
     for parsed_output_file in parsed_output_files:
         with open(parsed_output_file, 'r') as parsed:
             reader = csv.DictReader(parsed, delimiter="\t")
             for row in reader:
                 index = row['Index']
-                allele = row['HLA Allele']
+                if index in indexes_with_good_binders:
+                    continue
 
+                allele = row['HLA Allele']
                 score = float(row['Best MT Score'])
                 if args.allele_specific_binding_thresholds:
                     threshold = PredictionClass.cutoff_for_allele(entry[allele])
                     threshold = float(args.binding_threshold) if threshold is None else float(threshold)
                 else:
                     threshold = float(args.binding_threshold)
                 if score < threshold:
+                    indexes_with_good_binders.append(index)
+
+    for parsed_output_file in parsed_output_files:
+        with open(parsed_output_file, 'r') as parsed:
+            reader = csv.DictReader(parsed, delimiter="\t")
+            for row in reader:
+                index = row['Index']
+                if index in indexes_with_good_binders:
                     continue
 
-                if index not in iedb_results:
-                    iedb_results[index] = {}
-                if allele not in iedb_results[index]:
-                    iedb_results[index][allele] = {}
-                if 'min_score' in iedb_results[index][allele]:
-                    iedb_results[index][allele]['min_score'] = min(iedb_results[index][allele]['min_score'], score)
+                allele = row['HLA Allele']
+                score = float(row['Best MT Score'])
+                if index in min_scores:
+                    min_scores[index] = min(min_scores[index], score)
                 else:
-                    iedb_results[index][allele]['min_score'] = score
-                epitopes.append(row['MT Epitope Seq'])
-    return (iedb_results, epitopes)
+                    min_scores[index] = score
+    return min_scores
 
 def create_graph(iedb_results, seq_tuples):
     Paths = nx.DiGraph()
     spacers = [None, "HH", "HHC", "HHH", "HHHD", "HHHC", "AAY", "HHHH", "HHAA", "HHL", "AAL"]
     for ep in seq_tuples:
         ID_1 = ep[0]
         ID_2 = ep[1]
-        Paths.add_node(ID_1)
-        Paths.add_node(ID_2)
         for space in spacers:
             if space is None:
                 key = str(ID_1 + "|" + ID_2)
             else:
                 key = str(ID_1 + "|" + space + "|" + ID_2)
```

Comment:
totally minor point - might be better to use "for spacer in spacers", as I briefly thought this was a literal space character for some reason